.github/workflows/semantic-release.yaml

```diff
@@ -1,32 +1,82 @@
 name: Semantic Release
 
 on:
-  push:
-    branches:
-      - staging
-      - production
+  workflow_dispatch:
+    inputs:
+      environment:
+        type: choice
+        description: 'Environment to release to. Select "staging" for a pre-release or "production" for a full release.'
+        options:
+          - staging
+          - production
+        required: true
 
 jobs:
+  # Create a release
   create-release:
     runs-on: ubuntu-latest
+    outputs:
+      published: ${{ steps.release.outputs.published }}
+      tag: ${{ steps.release.outputs.tag }}
     steps:
-      - name: checkout-repo
+      - name: Checkout repo
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
           fetch-tags: true
           persist-credentials: false
           token: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
-      - name: checkout-actions
+      - name: Checkout actions
         uses: actions/checkout@v4
         with:
           repository: kiva/github-actions
           token: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
           persist-credentials: false
           path: .github/
-      - name: semantic-release
+      - name: Run semantic-release
+        id: release
         uses: ./.github/actions/semantic-release
         env:
           GITHUB_PAT: ${{ secrets.KIVA_ROBOT_GITHUB_PAT }}
         with:
           config: "node"
+
+  # Build and push the image to ECR if the release was published
+  build-image:
+    runs-on: ubuntu-latest
+    needs: create-release
+    if: needs.create-release.outputs.published == 'true'
+    env:
+      GITHUB_PAT: "${{ secrets.KIVA_ROBOT_GITHUB_PAT }}"
+      AWS_REGION: "us-west-2"
+      AWS_ACCESS_KEY_ID: "${{ secrets.AWS_ACCESS_KEY_ID }}"
+      AWS_SECRET_ACCESS_KEY: "${{ secrets.AWS_SECRET_ACCESS_KEY }}"
+      SLACK_BOT_TOKEN: "${{ secrets.SLACK_TOKEN }}"
+      SLACK_CHANNEL: "eng-build-failures"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ needs.create-release.outputs.tag }}
+      - uses: actions/checkout@v3
+        with:
+          repository: kiva/github-actions
+          token: ${{ env.GITHUB_PAT }}
+          path: .github/
+      - uses: snow-actions/sparse-checkout@v1.2.0
+        with:
+          repository: kiva/marketplace-web-ui-ci
+          token: ${{ env.GITHUB_PAT }}
+          path: .docker
+          ref: main
+          patterns: |
+              resources/org/kiva/marketplaceWebUiCi/ui
+      - name: move files
+        run: |
+          mv .docker/resources/org/kiva/marketplaceWebUiCi/ui/Dockerfile .
+      - name: docker-build
+        uses: ./.github/actions/docker-build-push
+        if: github.event_name == 'pull_request'
```

Comment:
When someone triggers a manual build, will the Github event name be `pull_request`? Or was this moved over from elsewhere and needs to be updated, perhaps?

Reply:
Good catch! That needs to be updated